renku/core/management/repository.py

```diff
@@ -174,6 +174,38 @@ def project(self):
         """Return the FOAF/PROV representation of the project."""
         return Project.from_yaml(self.renku_metadata_path, client=self)
 
+    @property
+    def remote(self, remote_name='origin'):
+        """Return host, owner and name of the remote if it exists."""
+        from renku.core.models.git import GitURL
+
+        host = owner = name = None
+        try:
+            remote_branch = \
+                self.repo.head.reference.tracking_branch()
+            if remote_branch is not None:
+                remote_name = remote_branch.remote_name
+        except TypeError:
+            pass
+
+        try:
+            url = GitURL.parse(self.repo.remotes[remote_name].url)
+
+            # Remove gitlab. unless running on gitlab.com.
+            hostname_parts = url.hostname.split('.')
+            if len(hostname_parts) > 2 and hostname_parts[0] == 'gitlab':
+                hostname_parts = hostname_parts[1:]
```

Comment:
I don't get what this part is doing. Why `gitlab` should be removed?

Reply:
this is a consequence of how gitlab is configured in renku deployments - it's not on a subdomain but instead on a path e.g. `renkulab.io/gitlab`. But the host should not include `gitlab`.

Reply:
If `gitlab` is part of the path (and not part of the hostname) then this won't do anything. In this case, for example, `hostname` will be `renkulab.io`.

Reply:
hmm you're right, I'm probably misremembering what the point was here, I have to look at the original commit that introduced this. It actually looks to be the opposite from what I described above, i.e. it's removing just the subdomain part. So in the case that the remote is `gitlab.renkuinstance.com` we don't want the IDs to have `gitlab` in them but just `renkuinstance.com`